Add code of conduct for the team

malvikasharan opened the pull request:
# Summary

- Adding Code of Conduct for the team.
- Ask all members to review and leave a comment if they are happy to have this Code of Conduct added to the project
- Need the restorative justice element to be discussed and reviewed with @KirstieJane

KirstieJane commented:
I really love the focus on restorative justice!

My only very high level feedback at the moment is that I think we need to be clear about how the code of conduct interacts - or where it sits in a pipeline - with Turing HR processes. (Or - if we make it more general, HR processes at any organisation.)

Probably my suggestion is that the code of conduct should be norm setting within a team _before_ any involvement of HR. Then - if an HR process, including an informal one, is started, that one needs to take precident over code of conduct enforcement. It's a bit sad because although HR will always _try_ for a restorative justice position they may not take the same steps "we" would if we were handling a complaint through the code of conduct alone. So being really clear where the boundaries are is the best - if imperfect - way forwards.

malvikasharan commented:
@KirstieJane I have included your comment with appropriate Turing HR policy links in the claus 15.2. I will get the CM team review this and apply to this repo to avoid a near CoC breach incident that could occur on this repo (commenting blocked on issue by me).

RaoOfPhysics commented:
Morning, @malvikasharan. I gave this a read yesterday and am happy from my side for it to be merged.

arronlacey commented:
agree with @RaoOfPhysics - happy for this to be merged.

malvikasharan commented:
@arronlacey Apology for not addressing on of your points in #104 
https://github.com/alan-turing-institute/open-community-building/pull/104#issuecomment-1196752410

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,333 @@
+1. An imposed suspension, such as asking reportee to “take a week or month off” from the GitHub repo or team activity.
+1. recommended training activity relevant to correcting the behaviour or building awareness/understanding required to right the wrongs.
+1. a stronger warning with consequences for continued behaviour. 
+
```
@arronlacey should we add something here to address what if a resolution can't be agreed (from #104).

Assumption is that people involved in report handling will arrive to some kind of resolution. However, we can include that if resolution is not reached, the ombudsperson will be cconsulted and any delay will be communicated with the reporter and reportee.

malvikasharan replied:
```suggestion
1. if a resolution is not reached, the ombudsperson will be consulted and any delay or status update for the CoC enforcement will be communicated with the reporter and reportee.
```

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,333 @@
+
+To report an issue involving the team lead, please email directly to the Tools, Practices and Systems 
+Programme Director, Kirstie Whitaker. Since she is currently on maternity leave, please contact TPS
+Programme Manager Arielle Bennett (abennett@turing.ac.uk) - who is also our ombudsperson. 
```
@arielle-bennett thanks for agreeing to be added in the CoC. Can you please look at the CoC and the details that this role entails?

malvikasharan replied:
Approved by Arielle personally.

malvikasharan commented:
@EKaroune and @arronlacey have agreed to included as CoC contact in this document.

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+Community leaders will reach out to the reporter to ask for further information if the submitted report is not complete or needs some additional context.
+Community leaders will also ensure that person impacted or harmed by the reported incident is safe and identify how they can be further supported.
+
+**15.1.1** A meeting with the reportee will be scheduled where the person handling the report will discuss four questions:
```
```suggestion
**15.1.1** A meeting with the reportee will be scheduled where the person handling the report will discuss five questions:
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+If a participant does not abide by the Code of Conduct, we will use the 
+“restorative practices” approach to handle the situation. Restorative Practices has 
+their roots in “restorative justice,” a way of looking at wrongdoing that focuses on 
+repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. 
```
```suggestion
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoers. 
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+
+- 4.2 **Be welcoming**: we strive to build a welcoming and supportive team.
+  As community facilitators, we also strive to apply this value in creating 
+  Communities of Practice that welcomes and support members of all backgrounds 
```
```suggestion
  Communities of Practice that welcomes and supports members of all backgrounds 
```

aleesteele replied:
Is it 'welcome and support' since communities of practice is plural?

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+  professionals, and we conduct ourselves professionally. Be kind to others. Do
+  not insult or put down other participants. Harassment and other exclusionary
+  behaviour aren't acceptable. This includes, but is not limited to Violent
+  threats or language directed against another person, Discriminatory jokes and
```
```suggestion
  threats or language directed against another person, discriminatory jokes and
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+  not insult or put down other participants. Harassment and other exclusionary
+  behaviour aren't acceptable. This includes, but is not limited to Violent
+  threats or language directed against another person, Discriminatory jokes and
+  language, Posting sexually explicit or violent material, Posting (or
```
```suggestion
  language, posting sexually explicit or violent material, posting (or
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+  threats or language directed against another person, Discriminatory jokes and
+  language, Posting sexually explicit or violent material, Posting (or
+  threatening to post) other people’s personally identifying information
+  (“doxing”), Personal insults, especially those using racist or sexist terms,
```
```suggestion
  (“doxing”), personal insults, especially those using racist or sexist terms,
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+  language, Posting sexually explicit or violent material, Posting (or
+  threatening to post) other people’s personally identifying information
+  (“doxing”), Personal insults, especially those using racist or sexist terms,
+  Unwelcome sexual attention, Advocating for or encouraging, any of the above
```
```suggestion
  unwelcome sexual attention, advocating for or encouraging, any of the above
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+  threatening to post) other people’s personally identifying information
+  (“doxing”), Personal insults, especially those using racist or sexist terms,
+  Unwelcome sexual attention, Advocating for or encouraging, any of the above
+  behaviour, and Repeated harassment of others. In general, if someone asks you to
```
```suggestion
  behaviour, and repeated harassment of others. In general, if someone asks you to
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+2. Prioritises persons affected and the consequent needs of victims and safety in communities.
+3. Addresses obligations of the wrongdoer resulting from those harms, as well as the team/community’s obligations to both victims and wrongdoers.
+4. Creates opportunity for all individuals in this team, who have a legitimate stake in a given situation, to be heard and their concerns addressed.
+5. Seeks to right the wrongs that have been done so that victims feel safe and valued, and wrongdoers feel restored to the team/community
```
```suggestion
5. Seeks to right the wrongs that have been done so that victims feel safe and valued, and wrongdoers feel restored to the team/community.
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+2. Must establish policies and procedures that handle incident reports in a way that strengthens relationships.
+3. Focuses on harms done rather than rules broken.
+4. Gives voice to the person who has been harmed.
+5. Engages in collaborative problem solving
```
```suggestion
5. Engages in collaborative problem solving.
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+4. Gives voice to the person who has been harmed.
+5. Engages in collaborative problem solving
+6. Empowers change and growth for all involved.
+7. Enhances responsibility for actions and attitudes of all involved
```
```suggestion
7. Enhances responsibility for actions and attitudes of all involved.
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+1. a private communication from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behaviour was inappropriate.
+1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report
```
```suggestion
1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report.
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+clarity around the nature of the violation and an explanation of why the
+behaviour was inappropriate.
+1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report
+1. an action plan for the reportee to improve their awareness and correct future behaviours
```
```suggestion
1. an action plan for the reportee to improve their awareness and correct future behaviours.
```

vhellon commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,336 @@
+behaviour was inappropriate.
+1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report
+1. an action plan for the reportee to improve their awareness and correct future behaviours
+1. reportee will have no interaction with the reporter, mediating interaction between reporter and reportee is not in scope of this Code of Conduct
```
```suggestion
1. reportee will have no interaction with the reporter, mediating interaction between reporter and reportee is not in scope of this Code of Conduct.
```

vhellon reviewed (commented):
All looks good to me- have just corrected some typos

RaoOfPhysics commented:
LGTM! 🚀

aleesteele commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+> _A code of conduct is a set of rules outlining the norms, rules, and responsibilities or proper practices of an individual party or an organization._
+
+This Code of Conduct document outlines our expectations for 
+participants within the team as well as this online repository (project).
```
```suggestion
participants within the team as well as this online repository (and the ideas, projects, and collaborations that might emerge from it).
```
Just adding this for clari

Arielle-Bennett commented:
This all looks good to me, happy to be ombudsperson for the CoC ✨

malvikasharan commented:
@vhellon @aleesteele @RaoOfPhysics @Arielle-Bennett @arronlacey Thank you all for review and comments. 🎊

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+1. recommended training activity relevant to correcting the behaviour or building awareness/understanding required to right the wrongs.
+1. a stronger warning with consequences for continued behaviour. 
+1. if a resolution is not reached, the ombudsperson will be consulted and any delay or status update for the CoC enforcement will be communicated with the reporter and reportee.
+**16.1** After the Resolution is Agreed Upon
```
```suggestion

**16.1** After the Resolution is Agreed Upon
```

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+- [Research Ethics Policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2241&SearchId=0&utm_source=interact&utm_medium=general_search&utm_term=code+of+conduct)
+- [Data protection policy](https://mathison.turing.ac.uk/Interact/Pages/Content/Document.aspx?id=2184&SearchId=23865&utm_source=interact&utm_medium=general_search&utm_term=data+protection)
+
+### **15.2.2** If the reportee is not the Turing Employee
```
```suggestion
### **15.2.2** If the reportee is not a Turing Employee
```

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,338 @@
+Code of Conduct should be considered as a norm-setting mechanism for the Open Research Community Building Team before any involvement of HR in the interactions that take place within the team. 
+
+If the reportee is a Turing employee, for instance, Community Managers and senior researchers, and the incident is considered out of the scope of this Code of Conduct due to the severity of the matter/case, an HR process (formal or informal), will be started.
+In this case, the report will be passed to the HR and/or appropriate personnel under the Turing policies and their policy related enforcement mechanism will take precedent over the Code of Conduct enforcement.
```
```suggestion
In this case, the report will be passed to the HR and/or appropriate personnel under the Turing policies and their policy related enforcement mechanism will take precedent over this Code of Conduct enforcement.
```

aleesteele commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+We want to establish a Code of Conduct that focuses on creating a welcoming and safe environment for
+everyone involved.
+If a participant does not abide by the Code of Conduct, we will use the 
+“restorative practices” approach to handle the situation. Restorative Practices has 
```
```suggestion
“restorative practices” approach to handle the situation. Restorative practices have
```

aleesteele commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+If a participant does not abide by the Code of Conduct, we will use the 
+“restorative practices” approach to handle the situation. Restorative Practices has 
+their roots in “restorative justice,” a way of looking at wrongdoing that focuses on 
+repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. 
```
```suggestion
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. This approach emerged from studies of and in reaction to incarceration and punishment within criminal justice systems, and has its roots in indigenous belief systems.
```
I suggest this addition for greater context & intellectual lineage, especially for those that might not be familiar with the approach.

malvikasharan replied:
Slight rewording suggestion for this context:
> Although the 'restorative justice' term was in 1970 in the context of social work and justice systems, restorative practice has its roots in indigenous practices employed in cultures all over the world.

aleesteele commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+1. Demonstrating empathy and kindness toward other people
+1. Being respectful of differing opinions, viewpoints, and experiences
+1. Gracefully giving and accepting constructive feedback
+1. Accepting responsibility for harm done to people/community, correcting our mistakes,
```
```suggestion
1. Accepting responsibility for harm done to people and/or communities, correcting our mistakes,
```

aleesteele commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+**5.1** Examples of behaviour that contributes to a positive environment for our
+community include:
+
+1. Demonstrating empathy and kindness toward other people
```
The numbers are displayed as a series of 1's, not sure why on my end.

sgibson91 replied:
This is a markdown "hack". If you look at the rendered copy, you'll see that the numbers progress as you'd expect. This means that authors don't have to keep track of how many bullets they have, and can also help mitigate issues where instructions might say "refer to step 3", but then a new PR adds a new step before so step 3 isn't step 3 anymore, etc.

malvikasharan replied:
Thanks Sarah for addressing this!

aleesteele replied:
Thanks so much @sgibson91 for the clarification! 👍

aleesteele commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,337 @@
+1. Being respectful of differing opinions, viewpoints, and experiences
+1. Gracefully giving and accepting constructive feedback
+1. Accepting responsibility for harm done to people/community, correcting our mistakes,
+  and learning from the experience to avoid future harm
```
I wonder if it would be worth adding in quotes: 
```suggestion
  and learning from the experience to avoid future harm (following the 'regenerative justice' framework)
```

malvikasharan replied:
I would avoid adding new concept unless strictly necessary. Has this term been mentioned before? If not, do we need to mention it at all?

aleesteele replied:
I agree! Especially with the addition of the line above about regenerative justice, I don't think this is a necessary addition anymore.

sgibson91 commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,338 @@
+
+## 9. Restorative Disciplines
+
+1. Acknowledges that trust, safe space and relationships are central to community building.
```
```suggestion
1. Acknowledges that trust, safe spaces and relationships are central to community building.
```

?

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,338 @@
+  As community facilitators, we also strive to apply this value in creating 
+  Communities of Practice that welcomes and supports members of all backgrounds 
+  and identities. This includes, but is not limited to
+  members of any race, ethnicity, culture, national origin, colour, immigration
```
The 'commitment' section is supposed to be exhaustive. This one is not exhaustive, instead we highlight the top protected characteristics. This can be kept identical if you suggest.

eirini-zormpa replied:
No, that's okay! I would maybe just remove "colour" from this list as I think it doesn't add anything to race and reads as perhaps older terminology? Do let me know if I'm missing something though 😊

malvikasharan replied:
Now that inconsistency is definitely bothering me. Let me suggest some consistent change.

malvikasharan replied:
I have merged "our commitment" with "diversity statement" into single clause 2: https://github.com/alan-turing-institute/open-community-building/blob/code-of-conduct/CODE_OF_CONDUCT.md#2-our-commitment-and-diversity-statement

To specifically address the inconsistency issue, I have now updated 4.2 that references clause 2.
Let me know if this actually works better.

eirini-zormpa replied:
I like that!

malvikasharan replied:
Great! Thank you Eirini.

malvikasharan commented:
Thanks @aleesteele @eirini-zormpa and @sgibson91 for the review.

@aleesteele and @eirini-zormpa, there are some unresolved comments. Can you please have a look when you have time next week? Especially to check if my response addresses your comments/suggestions.

malvikasharan commented:
Hi @aleesteele can you please respond to the unresolved comments?

malvikasharan commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,342 @@
+fairly and equally as possible. Whenever a member has made a mistake, we
+expect them to take responsibility for it. If someone has been harmed or
+offended, it is our responsibility to listen carefully and respectfully, and do
+our best to right the wrong.
```
```suggestion
our best to right the wrong.

Although the 'restorative justice' term was coined in 1970 in the context of social work and justice systems, restorative practice has its roots in indigenous practices employed in cultures all over the world.
```

aleesteele replied:
Looks great!

malvikasharan commented:
I have verbal approval from Arron and Emma from our strategy meeting. Posting here for the record.

malvikasharan commented:
Merging this!! Thank you all! <3
